tests/baselines/reference/checkDashedJsxAttributes.js

```diff
@@ -0,0 +1,91 @@
+//// [checkDashedJsxAttributes.tsx]
+/// <reference path="/.lib/react16.d.ts" />
+
+// Repro from #44797
+
+import * as React from "react";
+
+interface Props {
+    foo: string;
+    bar: number;
+    [dataProp: `data-${string}`]: string;
+}
+
+function Yadda(props: Props) {
+    return <div />;
+}
+
+let props: Props = {
+    foo: "",
+    bar: 0,
+    "data-yadda": 42,
+};
+
+let x1 = <Yadda foo="hello" bar={42} data-yadda={42} />;
+
+
+
+let propsObj1: Props = {
+    foo: "",
+    bar: 0,
+    "data-yadda": 42,
+};
+
+// Should error on data-yadda
+let x2 = <Yadda foo="hello" bar={42} data-yadda={42} />;
+
+/////////
+
+let propsObj2 = {
+    foo: "",
+    bar: 0,
+    "data-yadda": 42,
+};
+
+// Should error on data-yadda
+let y = <Yadda {...propsObj2} />
+
+propsObj1 = propsObj2;
+
+//// [checkDashedJsxAttributes.js]
+"use strict";
+/// <reference path="react16.d.ts" />
+var __assign = (this && this.__assign) || function () {
+    __assign = Object.assign || function(t) {
+        for (var s, i = 1, n = arguments.length; i < n; i++) {
+            s = arguments[i];
+            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
+                t[p] = s[p];
+        }
+        return t;

```

Comment:
⚙️ (optional) If you want to make the code slightly more concise, you can remove some of the unnecessary variables:
```suggestion
        for (var i = 1; i < arguments.length; i++) {
            var source = arguments[i];
            for (var key in source) {
                if (Object.prototype.hasOwnProperty.call(source, key)) {
                    target[key] = source[key];
                }
            }
        }
        return target;
```
This version has the same functionality but with fewer variables. It's a matter of style preference, and both versions are valid.